python/nav/ipdevpoll/timestamps.py

```diff
@@ -103,7 +103,12 @@ def _deserialize():
                 return None
             try:
                 return json.loads(info.value)
-            except Exception:
+            except (
+                AttributeError,
+                json.JSONDecodeError,
+                UnicodeDecodeError,
+                TypeError,
+            ):
```

Comment:
Not sure I'm convinced about this particular change, unless `json.loads()` guarantees that these are the only exceptions it will ever raise.

The point of this handler is that if there is any reason at all that the stored timestamp cannot be read, then we discard it and treat it as though no timestamp was stored (which is sort of 'invalidating' a cache).  Callers do not expect an exception here.  An exception from this function would likely abort an ipdevpoll job for no good reason.  If the invalid value remains in the db, the job would never resume working until someone hand-removed the value from the database.  Whereas ignoring the exception would cause the data to be fully collected over again, and a new timestamp is stored.

Reply:
In the [documentation for json.load](https://docs.python.org/3/library/json.html#json.load) it states that is only raises `JSONDecodeError` and `UnicodeDecodeError`. The other two errors come from accessing `info.value`

Reply:
I'll buy what the docs say, then :)